packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -169,3 +182,16 @@ function mark_reactions(signal, status) {
 		}
 	}
 }
+
+export function enter_constructor() {
+	const prev = constructor_reaction;
+	constructor_reaction = current_reaction;
+	return prev;
+}
+
+/**
+ * @param {Reaction | null} reaction
+ */
+export function leave_constructor(reaction) {
+	constructor_reaction = reaction;
+}
```

Comment:
if a constructor throws could this become corrupted?

Reply:
That's why I had the `try` builder there. However, I think we can likely get away without needing it as the current derived will not match. I can put back the try/finally, but it does add quite a bit of code.